cache/common.go

```diff
@@ -0,0 +1,20 @@
+package cache
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"os"
+)
+
+// We need this prefix because there could be multiple restore steps in one workflow with multiple cache keys
+const cacheHitEnvVarPrefix = "BITRISE_CACHE_HIT__"
+
+func checksumOfFile(path string) (string, error) {
+	hash := sha256.New()
+	b, err := os.ReadFile(path)
```

Comment:
Do we have any guarantees about the size of the file?  This could cause a huge memory overhead if we're slurping large files.  Hashing of files is typically done by reading a frame of bytes from the file at a time.  I looked up the documentation and they provided a [good example](https://pkg.go.dev/crypto/sha256#New).  Looks like `io.Copy` handles those concerns for us!

Reply:
Good catch, thank you! I updated the implementation here and in the `keytemplate` package as well.